### PR TITLE
Fixed Bugs with Pause Menu and Dialogue Input

### PR DESCRIPTION
**Pull request by getmikyled**
BUGS FIXED:

- Cursor lock mode not returning to right state when pausing while dialogue is playing
- Fixed being able to advance dialogue while game is paused

### CODE REVIEW

File: Assets/Scripts/UI/PauseMenuUI.cs [hunk 1]
```diff
@@ -25,23 +27,30 @@ private void Update()
...
+                Resume(prevCursorLockMode);
             }
             else
             {
                 Pause();    
             }
         }
     }
-    public void Resume()
+    public void Resume(CursorLockMode cursorLockMode)
```
Comment (TheStress) [inline Assets/Scripts/UI/PauseMenuUI.cs]: Adding a parameter to this broke the pause menu prefab (the resume button)

File: Assets/Scripts/UI/PauseMenuUI.cs [hunk 2]
```diff
@@ -11,6 +11,8 @@ public class PauseMenuUI : MonoBehaviour
     private bool gameIsPaused = false;
     public KeyCode pauseKey;
 
+    private CursorLockMode prevCursorLockMode;
```
Comment (TheStress) [inline Assets/Scripts/UI/PauseMenuUI.cs:14]: you also need to keep track of the previous cursor visibility

File: Assets/Scripts/UI/PauseMenuUI.cs (hunk 1 above)
Comment (getmikyled) [reply Assets/Scripts/UI/PauseMenuUI.cs]: Fixed! I could not see anything in the pause menu UI prefab was testing it, I had assumed that it was not implemented yet.

File: Assets/Scripts/UI/PauseMenuUI.cs (hunk 2 above)
Comment (getmikyled) [reply Assets/Scripts/UI/PauseMenuUI.cs:14]: Fixed!

File: Assets/Scripts/UI/PauseMenuUI.cs [hunk 3]
```diff
@@ -50,7 +60,12 @@ public void Pause()
         pauseMenuUI.SetActive(true);
         Time.timeScale = 0f;
         gameIsPaused = true;
+        
+        playerInput.DisableInput();
         playerInput.DisableAbilityInput();
+
+        prevCursorVisibility = Cursor.visible;
+        prevCursorLockMode = Cursor.lockState;
```
Comment (TheStress) [inline Assets/Scripts/UI/PauseMenuUI.cs:68]: While testing this I found that the issue still exists so I swapped reading the previous cursor state and the assignments of the new cursor state.

**Review by TheStress — APPROVED** [review]
Looks good

File: Assets/Scripts/UI/PauseMenuUI.cs (hunk 3 above)
Comment (getmikyled) [reply Assets/Scripts/UI/PauseMenuUI.cs:68]: Ah sorry it did not work. I did not see any issues on my end. Weird fix though, I don't see any issues with the logic.

**Pull request merged**